### PR TITLE
Carousel: Remove redundant reference to `interval=false` from docs & tests

### LINKED ISSUE #36526: In 5.2.0-beta1 > different behavior of carousel data-bs-interval="false"
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

In 5.2.0-beta1 (as compared to 5.1.3), the carousel has different behavior when specified with data-bs-interval="false"

In 5.1.3, it disables auto scroll

In 5.2.0-beta1, it scrolls instantly

I believe this is a bug, as the docs still say that it should disable auto scroll https://getbootstrap.com/docs/5.2/components/carousel/#disable-touch-swiping

### Reduced test cases

https://codepen.io/comicsanscoder/pen/QWQVJbq

(switching js between 5.2.0-beta1 and 5.1.3 demonstrates the bug)

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

5.2.0-beta1

### PR DESCRIPTION
**Pull request by GeoSot**
closes #36526

### CODE REVIEW

**Review by julien-deramond — COMMENTED** [review]
From what I've tested the boolean is just interpreted as a number when the interval is set. So `data-bs-interval="false"` would be 0 (very quick) and `data-bs-interval="true"` would be 1 (very quick as well). `data-bs-interval="false"` isn't used anymore to block the autoplay so I agree with this modification and the fact that the boolean support should be removed.

Maybe consider some extra-modifications in this PR:
* Remove `data-bs-interval="false"` in `carousel.spec.js`
* Add something in the migration guide to help the users?

**Comment by GeoSot** [comment]
Thank you @julien-deramond  for your suggestions. I've removed the references from the test file.

I don't think we need to add any migration notice. Wishing I am correct, during the latest refactoring, I didn't change the functionality and the checks that are being inside.

**Review by julien-deramond — APPROVED** [review]
LGTM!

File: site/content/docs/5.2/components/carousel.md [hunk 1]
```diff
@@ -314,7 +314,7 @@ const carousel = new bootstrap.Carousel('#myCarousel')
 {{< bs-table >}}
 | Name | Type | Default | Description |
 | --- | --- | --- | --- |
-| `interval` | number | `5000` | The amount of time to delay between automatically cycling an item. If `false`, carousel will not automatically cycle. |
```
Comment (XhmikosR) [inline site/content/docs/5.2/components/carousel.md:317]: Shouldn't this read:

```markdown
| `interval` | number | `5000` | The amount of time to delay between automatically cycling an item. If `0`, carousel will not automatically cycle. |
```

Comment (julien-deramond) [reply site/content/docs/5.2/components/carousel.md:317]: Apparently if `0` the interval is just very short (no interval) so the animation is very quick. There's a cycle when `data-bs-ride` is set to `true` or `carousel`.

Comment (GeoSot) [reply site/content/docs/5.2/components/carousel.md:317]: I agree with Julien here. Are you ok with it @XhmikosR ? can I proceed?

**Pull request merged**